From 51fe5125c92969cf7de8bd7d06605c50d10c8ba7 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter@cs.unisa.edu.au>
Date: Fri, 16 Nov 2007 16:45:45 +1030
Subject: [PATCH] Don't flush buttons, release events can cause paste events.

---
 src/mouse.c |   11 +----------
 1 files changed, 1 insertions(+), 10 deletions(-)

diff --git a/src/mouse.c b/src/mouse.c
index 293af74..953f542 100644
--- a/src/mouse.c
+++ b/src/mouse.c
@@ -1854,26 +1854,17 @@ MouseConvert(InputInfoPtr pInfo, int first, int num, int v0, int v1, int v2,
 
 /**********************************************************************
  *
- * FlushButtons -- send button up events for sanity.
+ * FlushButtons -- reset button states.
  *
  **********************************************************************/
 
 static void
 FlushButtons(MouseDevPtr pMse)
 {
-
-    /* If no button down is pending xf86PostButtonEvent()
-     * will discard them. So we are on the safe side. */
-
     int i, blocked;
 
     pMse->lastButtons = 0;
     pMse->lastMappedButtons = 0;
-
-    blocked = xf86BlockSIGIO ();
-    for (i = 1; i <= 5; i++)
-	xf86PostButtonEvent(pMse->device,0,i,0,0,0);
-    xf86UnblockSIGIO (blocked);
 }
 
 /**********************************************************************
-- 
1.5.3.7

